torch/distributed/optim/named_optimizer.py

```diff
@@ -260,11 +260,28 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
 
         self._optimizer.load_state_dict(new_state_dict)
 
-    # pyre-ignore [2]
-    def add_param_group(self, param_group: Any) -> None:
-        raise NotImplementedError(
-            "add_param_group not supported yet and might be implemented soon."
+    def add_param_group(self, param_group: Mapping[str, Any]) -> None:
+        warnings.warn(
+            "This API is still in develop and it might not work as expected "
+            "under the distributed training scenario."
         )
+        assert isinstance(param_group, dict), "param group must be a dict"
+
+        params = param_group['params']
+        if isinstance(params, torch.Tensor):
+            param_group['params'] = [params]
+        else:
```

Comment:
Can we ensure that unittests cover both of these code paths?

Reply:
Sure.